proto/ca_service.proto

```diff
@@ -40,17 +40,8 @@ service IstioCAService {
 message Request {
   // PEM-encoded certificate signing request
   bytes csr_pem = 1;
-  NodeAgentCredentials node_agent_credentials = 2;
-}
-
-message NodeAgentCredentials {
-  oneof node_credential {
-    GcpInstanceCredential gcp_instance_credential = 1;
-  }
-}
-
-message GcpInstanceCredential {
-  bytes signed_instance_metadata = 1;
+  // opaque credential for node agent
+  bytes node_agent_credential = 2;
```

Comment:
Depends on whether control plane can route the request to the correct authenticator? I think we have three requirements here:

1. handler for the node agent request and workload request needs to be different
2. handler for requests in different environments needs to be different
3. pluggability so that developers can easily create new authenticator in the control plane and let requests be handled by the custom authenticator

In my opinion type inference is not an extensible way of doing this. Maybe we can have an additional string field to indicate what type of this request is. Let's discuss this in the issues.

Reply:
Sounds good. I think we agree that a credential type is necessary, otherwise the CA will not be able to match authenticator for the request :) I think a string probably makes more sense for the pluggability.
BTW, I don't quite understand 1), do we expect the CA to act differently to a CSR from node agent VS. workload?

Reply:
I think it should be different because the way CA validates `node_agent_credential` would be different for node agent and workload. For node agent it's not always a x509 cert.

Reply:
In the existing use cases we cover, the workload doesn't directly talk to CA. Do you want to cover that use case?

Reply:
I wasn't talking about this use-case. Node agent can request both node identity and workload identity. And if they are both requested using `HandleCSR` rpc function, CA needs to differentiate the two cases because authentication of the request will be different (one is native node-level identity and one is issued node identity).

So the question really is: how should node agent present its identity when requesting workload identity? If the node identity is added to the `grpc.DialOption` as client cert, then CA needs to condition on whether a client cert is presented in the grpc connection, which is not a good thing I think. Or the node identity can be used as `node_agent_credential` field and let a different authenticator verify the cert.

Reply:
I see. I think currently the "node identity" and "workload identity" are the same in the single tenant per VM use case. We will give them different identities in the future, but I'm not discussing this problem here.

The "credential type" I mentioned here is for differentiating the credentials from GCP, AWS or other credential providers. The CA may have multiple authenticators, and it needs to know which one to use.

Reply:
I am curious why change this node_agent_credential? If it is always bytes, how does we know which authenticator to use?

Reply:
Yes that's the intent! I agree we can leave that problem for later.